core/services/workflows/delegate_test.go

```diff
@@ -6,46 +6,135 @@ import (
-workflowOwner = "00000000000000000000000000000000000000aa"
-`,
+			"missing workflow field",
+			func() string {
+				id := "15c631d295ef5e32deb99a10ee6804bc4af1385568f9b3363f6552ac6dbb2cef"
+				owner := "00000000000000000000000000000000000000aa"
+				return testspecs.GenerateWorkflowSpec(id, owner, "").Toml()
+			},
+			false,
+		},
+
+		{
+			"null workflow",
+			func() string {
+				id := "15c631d295ef5e32deb99a10ee6804bc4af1385568f9b3363f6552ac6dbb2cef"
+				owner := "00000000000000000000000000000000000000aa"
+				return testspecs.GenerateWorkflowSpec(id, owner, "{}").Toml()
+			},
+			false,
+		},
+
+		{
+			"minimal passing workflow",
+			func() string {
+				id := "15c631d295ef5e32deb99a10ee6804bc4af1385568f9b3363f6552ac6dbb2cef"
+				owner := "00000000000000000000000000000000000000aa"
+				wf := `
+triggers: []
+consensus: []
+targets: []
+`
+				return testspecs.GenerateWorkflowSpec(id, owner, wf).Toml()
+			},
 			true,
 		},
+
+		// Taken from jobs controller test, as we want to fail early without a db / slow test dependency
+		{
+			"valid full spec",
+			func() string {
+				id := "15c631d295ef5e32deb99a10ee6804bc4af1385568f9b3363f6552ac6dbb2cef"
+				owner := "00000000000000000000000000000000000000aa"
+				workflow := `
+triggers:
+  - id: "mercury-trigger@1"
+    config:
+      feedIds:
+        - "0x1111111111111111111100000000000000000000000000000000000000000000"
+        - "0x2222222222222222222200000000000000000000000000000000000000000000"
+        - "0x3333333333333333333300000000000000000000000000000000000000000000"
+
+consensus:
+  - id: "offchain_reporting@2"
+    ref: "evm_median"
+    inputs:
+      observations:
+        - "$(trigger.outputs)"
+    config:
+      aggregation_method: "data_feeds_2_0"
+      aggregation_config:
+        "0x1111111111111111111100000000000000000000000000000000000000000000":
+          deviation: "0.001"
+          heartbeat: 3600
+        "0x2222222222222222222200000000000000000000000000000000000000000000":
+          deviation: "0.001"
+          heartbeat: 3600
+        "0x3333333333333333333300000000000000000000000000000000000000000000":
+          deviation: "0.001"
+          heartbeat: 3600
+      encoder: "EVM"
+      encoder_config:
+        abi: "mercury_reports bytes[]"
+
+targets:
+  - id: "write_polygon-testnet-mumbai@3"
+    inputs:
+      report: "$(evm_median.outputs.report)"
+    config:
+      address: "0x3F3554832c636721F1fD1822Ccca0354576741Ef"
+      params: ["$(report)"]
+      abi: "receive(report bytes)"
+  - id: "write_ethereum-testnet-sepolia@4"
+    inputs:
+      report: "$(evm_median.outputs.report)"
+    config:
+      address: "0x54e220867af6683aE6DcBF535B4f952cB5116510"
+      params: ["$(report)"]
+      abi: "receive(report bytes)"
+`
+				return testspecs.GenerateWorkflowSpec(id, owner, workflow).Toml()
+			},
+			true,
+		},
+
 		{
 			"parse error",
-			`
+			func() string {
+				return `
 invalid syntax{{{{
-`,
+`
+			},
 			false,
 		},
+
```

Comment:
I added a newline between each test case so its easier to visually see + navigate with "paragraph" movement commands supported in editors like vim